### DISCUSSION TITLE
Supporting the script resource page

### DISCUSSION THREAD

**Discussion by kidayasuo** [w3c/jlreq-d#62] [General]
([English version is here](https://github.com/w3c/jlreq-d/discussions/62#discussioncomment-10332328))

Richard (@r12a) さんから、jlreq-dがScript Resources pageのセクション構造を採用するか、またはそのページのセクションに対応するリンクターゲットをjlreq-dに作成することを検討できないかとの[依頼](https://github.com/w3c/jlreq/issues/438)がありました。

これについて[2024-08-13のJLReq TFミーティング](https://github.com/w3c/jlreq/discussions/441)で議論した。リソースページの該当する項目からリンクできるようにjlreq-dで考慮するのが適当であろうとの結論になりました。リソースページの各項目に対する議論の結果を下記に示します。

---- 
# スクリプト・リソース・ページ

## 2. スクリプトの概要

jlreq-d第2章にリンクできる。

## 3. リソース全体へのリンク

jlreq-d全体へのリンクをここに置く。

### 4.1 縦書きテキスト
> 縦書きのテキストの要件は満たされていますか？通常は横書きのスクリプトと縦書きを混在させた場合はどうでしょうか？縦書きと横書きで異なる文字を使用するための切り替えが必要ですか？ブラウザは縦書きの行内に短い横書きのテキスト（日本語のタテ中ヨコ）を期待通りにサポートしていますか？文字の向きや文字の方向性の順序は、必要に応じてサポートされていますか？

jlreq-dの縦書き組版の説明にリンクできる。ただし、縦書きは、横書き組版からの差分として記述しているので、日本語組版の横書きの基礎を知らずに縦書きを行えるわけではない。よって基礎として横書きの組版の記述にもリンクする必要があるだろう。また、フォントを説明する章の縦組用グリフの説明にも関係がある。

### 4.2 双方向テキスト
> このスクリプトが右から左に流れる場合、それを扱う際に問題がありますか？双方向テキストは十分にサポートされていますか？数字や表現についてはどうですか？UnicodeのBidiコントロールやHTMLのマークアップは、必要なサポートを提供していますか？方向性の異なる流れの分離に問題がありますか？

日本語に関係なし。

### 5.1 フォントとフォントスタイル
> フォントは認識可能な書体にどのようにグループ化されていますか？各書体はどのように使用されていますか？OpenType機能が必要ですか？その他のフォント関連の要件や問題はありますか？

第2章基礎編のフォントの説明、およびフォントの要件を記述する章にリンクできる
- 和字の構造：ボディ、字面。ベースラインという概念のないこと
- 日本語書体の説明：明朝、ゴシック、丸ゴシック、のみ
- 中国字形との違い：

### 5.2 文脈に基づくシェーピングとポジショニング
> 文字が使用される文脈に応じて、特別なグリフの形が必要ですか？ある状況でグリフが相互作用しますか？文脈に応じてダイアクリティカルマークや他のアイテムを特別に配置する要件はありますか？スクリプトには、基本となる文字に対して同じ場所を争う複数のダイアクリティカルマークがありますか？

フォントの要件を記述する章にリンクする
- 縦書き横書きで異なる文字
- 同じコードポイントでも、和文約物の形が違うこと
- 合成濁音は本来の文字というよりUnicodeのアーキテクチャの問題なのでサイドノート

### 5.3 筆記体テキスト
> このスクリプトが筆記体の場合（例えばアラビア語、ンコ語、シリア語など）、筆記体テキストの処理に関連する問題や必要な機能はありますか？単語の一部にマークアップやスタイルを適用すると、筆記体のリンクが切れますか？Unicodeのジョイナーと非ジョイナーの文字は期待通りに動作しますか？

現代日本語に関係なし。

### 5.4 文字形の傾き、ウェイト、イタリック
> イタリック化、ボルド化、斜体化などは関連していますか？イタリックフォントは正しい方向に傾いていますか？合成されたイタリック化に問題はありますか？ボルド化やイタリック化に関連する他の問題はありますか？一般的な適用可能性の仮定に関連する問題はありますか？

日本語にはこの単位でまとまった概念はない。関連する情報はあるので、下の記述をjlreq-dのどこかに置き、そこへリンク。
- 斜体という一般的なスタイルは日本語には存在しない。ディスプレー書体にはあるが例外的。ウェイトはある。
- 上記を合成する場合、文字幅を変えないことが重要。

### 5.5 大文字と他の文字変換
> 正書法はバイカメラル（二分形）ですか？特に二つの間で変換が必要な場合、他の文字のペアリングがありますか？

日本語にはこの単位でまとまった概念はなく、よってjlreq-dにも該当するセクションはないが、関連する情報は何箇所かにある。それらにリンクするか。
- jlreq-dで該当するのはルビの小書きの大書きへの変換くらいか。大書きにしたい場合でもアクセスビリティを考慮して小書でデータを作り、大書きに変換すべき。
- 縦書きにする際に、文字を全角化したい場合がある。それを述べるかどうかは未定。

### 6.1 文字とエンコーディング
> ほとんどの言語は現在Unicodeでサポートされていますが、依然としていくつかの問題があります。特に、文字の順序に関連する問題や、競合するエンコーディング（ミャンマーのように）、またはバリエーションセレクターやエンコーディングモデルの標準化（モンゴル語のように）に関連する問題があるかもしれません。

エンコーディングのトピックはjlreq-dの現在の計画にないが、作ることを検討する。
- Unicodeの扱いや、どのようなUnicodeの機能を実装している必要があるか、例えば、漢字や絵文字を始め16bitを超える部分にも文字がある、濁音合成、比較のための正規化、など。
- 和文約物U+2000の問題、つまりproportionalと和文のコードポイントが区別きず、縦組で正立なのかどうかの区別もつかない問題に触れる。似た問題はギリシャ文字にもある。

### 6.2 グラフィーム／単語のセグメンテーションと選択
> Unicodeのグラフィームクラスターは、このスクリプトの文字単位を適切にセグメント化していますか？以下の操作に特別な要件はありますか：前方／後方の削除、カーソル移動と選択、文字カウント、検索とマッチング、テキストの挿入、改行、均等割り付け、ケース変換、並べ替え？単語はスペースや他の文字で区切られていますか？テキストをダブルクリックしたときに特別な要件がありますか？単語はハイフネーションされますか？

動的な動作の要件を記述する章にリンク。また行の折り返しなどにも関係する。

### 7.1 フレーズとセクションの境界
> フレーズ、文章、およびセクションの境界を示すために使用される文字は何ですか？ダッシュ、接続子、区切り文字などの他の句読点についてはどうですか？

これは言語の書き方の慣習に (language writing conventions) 関する話題であって、レイアウトには関係ないのでjlreq-dの範囲外。

### 7.2 引用符と引用
> 引用を示すために使用される文字は何ですか？引用内の引用は異なる文字を使用しますか？会話を示すために使用される文字は何ですか？同じメカニズムが単語の引用、または引用符に使用されますか？本や記事の名前を引用する場合はどうですか？

これは言語の書き方の慣習に (language writing conventions) 関する話題であって、レイアウトには関係ないのでjlreq-dの範囲外。

引用符は横書き縦書きで異なるので、それについての記述は存在するが網羅的に述べているわけではない。

### 7.3 強調とハイライト
> 強調とハイライトはどのように実現されていますか？テキストの横に、上に、または貫通して線やマークが引かれる場合、それらはテキスト自体から特別な距離を保つ必要がありますか？下線を引くなどの際に文字をスキップすることが重要ですか？縦書きのテキストではどう変わりますか？

jlreq-d第3章に強調について述べたセクションあり。そこにリンクできる。

### 7.4 省略、エリプシス、および繰り返し
> 省略、エリプシス、および繰り返しを示すために使用される文字や他の方法は何ですか？

これは言語の書き方の慣習に (language writing conventions) 関する話題であって、レイアウトには関係ないのでjlreq-dの範囲外。

欧文エリプシスと和文三点リーダの位置の違いや、二つ続けることの説明はあるが、このトピックについて網羅的な説明はない。

### 7.5 インラインメモと注釈
> *インライン*メモや注釈を作成するために使用されるメカニズムは何ですか？（脚注などの参照型のメモについては、下記を参照してください。）

jlreq-dに注のセクションあり。そこにリンクできる。インライン形式、およびページの構造に依存しないサイドノートなどについてはカバーするが、ページの存在が必要な注の形式は範囲外。

### 7.6 テキストデコレーションと他のインライン機能
> 下線、数値のオーバーバーなどのようなテキストの他の形態のハイライトやマーキング。テキストの範囲に関する情報を伝えるために使用される文字や方法（例：テキストデコレーション）は何ですか？テキストの横に、上に、または貫通して線が引かれる場合、それらはテキスト自体から特別な距離を保つ必要がありますか？下線を引くなどの際に文字をスキップすることが重要ですか？縦書きのテキストではどう変わりますか？前述のセクションで取り上げられなかった他の句読点はありますか？

jlreq-dではtext decorationを強調かどうかによって分けていない。そもそも強調が目的でないデコレーションは日本語にはないと思われる。よって強調のセクションにリンク。

### 7.7 データフォーマットと数字
> スクリプトに独自の数字セットがある場合、それらの使用方法に問題がありますか？スクリプトや言語は、問題を引き起こす特別なフォーマットパターンを使用しますか（例：インドの12,34,000）？日付/時刻フォーマットと選択についてはどうですか？非グレゴリオ暦が必要ですか？数字に関連するパーセント記号や他のシンボルは正しく動作し、数字に特別なデコレーション（エチオピア語やシリア語のように）が必要ですか？ウェブページでの個人名や住所の管理についてはどうですか：問題はありますか？

これは言語の書き方の慣習に (language writing conventions) 関する話題であって、レイアウトには関係ないのでjlreq-dの範囲外。

### 8.1 改行とハイフネーション
> テキストが行の末端に達したときに、特別な折り返しルールがありますか？改行は単語全体を折り返しますか、それとも文字や他のもの（チベット語やジャワ語のような音節など）ですか？行の最後や先頭に現れるべきではない文字があり、それを防ぐためにはどうすればよいですか？ハイフネーションが使用されますか、または他のものが使用されますか？どのようなルールが使用されていますか？どのような困難がありますか？

行の折り返しは現在単独のセクションになっていないが、その話題を独立させて、そこにリンク。

### 8.2 テキストの整列と均等割り付け
> 段落内のテキストは両側に均等なラインを持つ必要がありますか？スクリプトは、行の先頭や末尾で句読点がテキストボックスの外に出ることを許可しますか？ラインを均等にするために調整が必要な場合、それはどのように行われますか？スクリプトは単語間や文字間のスペースを縮めたり伸ばしたりしますか？アラビア語のように単語のベースラインを伸ばしますか？段落のインデントはどうですか？

該当するセクションにリンクできる。おそらくこの塊ではなく、複数のセクションに分かれる。

### 8.3 テキストの間隔
> このセクションでは、均等割り付けの際に導入されるものを超えて、文字間に適用される間隔について検討します。たとえば、正書法は単語、文字、または音節をスペースを広げることで強調や他の効果を生み出しますか？（均等割り付けに関連する間隔については、上記の「テキストの整列と均等割り付け」を参照してください。）

jlreq-dにそのセクションを作ってそこにリンク。

### 8.4 ベースライン、行の高さなど
> スクリプトには、混在するスクリプト間や一般的なベースラインの整列に特別な要件がありますか？このスクリプトには特別な行の高さがありますか？行間隔や行内の項目の垂直方向の位置に影響を与える他の側面はありますか？

jlreq-dのおそらく二箇所になる。混植のセクション、および行間について説明するセクション。

スクリプトが混在する場合、ベースラインの位置のみではなく、相対的なポイントサイズの調整も必要になる。それはScript Resoruce pageでは触れないのか？

### 8.5 リスト、カウンターなど
> リストや他のカウンタースタイルが使用されていますか？もしそうなら、その形式は何ですか？カウンターは縦書きで直立する必要がありますか？カウンターやリストに関連する他の側面はありますか？

JLReqではカウンター文字については触れていないが、jlreq-dではカバーすることを検討する。作るならそこにリンク。例えば縦書きのリストのカウンターは必ず成立させる。

### 8.6 頭文字のスタイリング
> スクリプトは、ドロップキャップや類似のための段落や行の最初の文字を特別にスタイリングしますか？大きな文字と隣接する行とのサイズの関係はどうですか？大きな文字は隣接する行に対してどこにアンカーされますか？大きな文字に引用符を含めるのは通常ですか？大きな文字は本当に音節ですか？落下、沈降、そして浮き上がりのタイプは見られますか？など。

日本語に関係なし。

## 9. ページと書籍のレイアウト

jlreq-dではページや書籍を扱わないので対象外。ただし、ページに依存しない項目がここに含まれる。

### 9.1 一般的なページレイアウトと進行
> メインのテキストエリアと補助エリアはどのように配置され定義されていますか？ここに特別な要件がありますか（日本語の基本版面の文字数など）？右から左に読むスクリプトの書籍のカバーは、背表紙の右側にあります。コンテンツが縦に、または左右に流れることができる場合、オブジェクト、テキストなどの位置を流れに対して指定する方法はどうしますか？縦書きのテキストでは列はどのように機能しますか？縦書きと横書きのブロックを混在させることができますか？テキストのスクロールは異なる方向に進みますか？

### 9.2 グリッドと表
> スクリプトには特別な文字グリッドや表の要件がありますか？

### 9.3 脚注、エンドノートなど
> 脚注、エンドノート、またはこの種の他の必要な注釈のサポートは、他の文化では異なる場合があります。場合によっては、スクリプトは注釈を配置するか、テキストからそれらにリンクするための非常に特異なアプローチを使用する場合があります。（インライン注釈のためのメモと注釈については、「インラインメモと注釈」を参照してください。このセクションは、参照マークと注釈の内容を分離するシステムに関するものです。）

インラインでない、しかしpaged mediaにも依存しない注、例えばサイドノートはScript Resource pageのどこでカバーする？

### 9.4 ページヘッダー、フッターなど
> ページ番号付けやランニングヘッダーの処理に特別な慣習がありますか？

### 9.5 フォームとユーザーインタラクション
> 縦書きのフォームコントロールが必要ですか？スクロールバーが異常な位置にありますか？ユーザーインタラクションに関する他の特別な要件はありますか？

jlreq-dの動的要素の章にリンクできる。
これはpaged mediaであることに依存しないので、9章にある理由が不明。

//

**Comment by kidayasuo** [comment]
(This is English translation)

Richard (@r12a) filed an [issue](https://github.com/w3c/jlreq/issues/438) to see if jlreq-d could either adopt the section structure of the Script Resources page or consider creating link targets in jlreq-d that correspond to sections of the resource page.

This was discussed at the [JLReq TF meeting on 2024-08-13](https://github.com/w3c/jlreq/discussions/441). It was concluded that it would be appropriate for jlreq-d to consider creating links from the relevant sections of the Script Resources page. Below are the results of the discussion for each item on the resource page.

---- 

# Script Resources Page

## 1. Introduction
## 2. Script Overview

Can link to Chapter 2 of jlreq-d.

## 3. All topics

A link to the entire jlreq-d should be placed here.

## 4. Text direction

### 4.1 Vertical text
Links can be made to the description of vertical text composition in jlreq-d. However, since vertical text is described as a differential from horizontal text composition, one cannot work with vertical text without knowledge of the basics of horizontal composition in Japanese typography. Therefore, it is also necessary to link to the description of horizontal composition as a foundation. Additionally, it relates to the description of vertical composition glyphs in the chapter on fonts.

### 4.2 Bidirectional text
Not relevant to Japanese language.

## 5. Glyph shaping and positioning

### 5.1 Fonts and font styles
Links can be made to the explanation of fonts in the basic section of Chapter 2, and the chapter describing font requirements:
- Structure of Japanese characters: body, character face. No concept of a baseline.
- Explanation of Japanese typefaces: Mincho, Gothic, Rounded Gothic only.
- Differences from Chinese glyphs.

### 5.2 Context-based shaping & positioning
Links can be made to the chapter describing font requirements:
- Different characters for vertical and horizontal writing.
- The shape of Japanese punctuation differs even with the same code point.
- Composite voiced sounds are more of a Unicode architecture issue than a native character concept, so it will be noted as a side note.

### 5.3 Cursive Text
Not relevant to modern Japanese language.

### 5.4 Letterform slopes, weights, & italics
Japanese language does not have a unified concept under this heading. However, since related information exists, it should be placed somewhere in jlreq-d with a link to it:
- Italics as a general style do not exist in Japanese typography. They do exist in display typefaces, but as an exception. Weight variations do exist.
- When synthesizing the above, it is important not to change the character width.

### 5.5 Case & other character transforms
Japanese language does not have a unified concept under this heading, so there is no corresponding section in jlreq-d, but related information exists in several places. Links could be made to those:
- The relevant section in jlreq-d might be the one on ruby text transformations, converting small kana to larger kana. Even when using larger kana, for accessibility reasons, data should be created in small kana and then converted.
- There are cases where characters need to be converted to full-width or to specify upright when using vertical text.

## 6. Typographic units
### 6.1 Characters & encoding
Encoding topics are not currently planned for jlreq-d but should be considered for inclusion:
- Handling of Unicode and necessary Unicode features implementation, e.g., characters beyond 16-bit, such as kanji and emojis, voiced sound composition, normalization for comparison.
- Issues like the U+2000 Japanese punctuation problem, i.e., the inability to distinguish between Latin and Japanese characters, and whether a character is upright in vertical text. Similar issues exist with Greek characters.

### 6.2 Grapheme/word segmentation and selection
Links can be made to the chapter describing requirements for dynamic behavior, as well as topics like line wrapping.

## 7. Punctuation & inline features
### 7.1 Phrase and section boundaries
This is a topic related to language writing conventions, not layout, so it is outside the scope of jlreq-d.

### 7.2 Quotations & citations
This is a topic related to language writing conventions, not layout, so it is outside the scope of jlreq-d.

There is a description of different quotation marks for horizontal and vertical text, but it is not exhaustive.

### 7.3 Emphasis & highlighting
There is a section on emphasis in Chapter 3 of jlreq-d that can be linked to.

### 7.4 Abbreviation, ellipsis & repetition
This is a topic related to language writing conventions, not layout, so it is outside the scope of jlreq-d.

There is an explanation of the difference between Western ellipses and Japanese three-dot leaders, and using two in sequence, but not a comprehensive explanation of this topic.

### 7.5 Inline notes & annotations
There is a section on annotations in jlreq-d that can be linked to. It covers inline format and non-inline annotations such as side notes that are independent of page structure.

### 7.6 Text decoration & other inline features
jlreq-d does not differentiate between text decoration and emphasis. It is believed that text decoration not intended for emphasis does not exist in Japanese layout. Therefore, link to the emphasis section.

### 7.7 Data formats & numbers
This is a topic related to language writing conventions, not layout, so it is outside the scope of jlreq-d.

## 8. Line & paragraph layout
### 8.1 Line breaking & hyphenation
Line wrapping is not currently a standalone section but could be separated out, with a link to it.

### 8.2 Text alignment & justification
Links can be made to the corresponding section. It will likely be split into several sections rather than grouped together.

### 8.3 Text spacing
Create a section in jlreq-d for this topic and link to it.

### 8.4 Baselines, line height, etc.
This will likely be covered in two sections of jlreq-d: the mixed script section and the section describing line spacing.

When scripts are mixed, not only the baseline position but also relative point size adjustments are necessary. Does the Script Resource page address this?

### 8.5 Lists, counters, etc.
JLReq does not cover counter characters, but jlreq-d will consider doing so. If created, links will be made there. For example, the counter in vertical lists must always be maintained.

### 8.6 Styling initials
Not relevant to Japanese language.

## 9. Page & book layout

Since jlreq-d does not cover pages or books, this section is out of scope. However, items that do not depend on pages are included here. see below.

### 9.1 General page layout & progression
### 9.2 Grids & tables
### 9.3 Footnotes, endnotes, etc
If the Script Resource page covers non-inline, but also not page-dependent notes, such as side notes, where is it discussed?

### 9.4 Page headers, footers, etc
### 9.5 Forms & user interaction
Links can be made to the dynamic elements chapter of jlreq-d.
Since this does not depend on paged media, why is it in Chapter 9?

**Comment by xfq** [comment]
See also the clreq discussion: https://www.w3.org/2024/08/07-clreq-minutes.html#t02

**Comment by r12a** [comment]
a few observations...

> 5.4 Letterform slopes, weights, & italics

I think there are some topics where it is useful to say that a typographic feature is not used – and italicisation is one such.  Most western developed applications now provide 'italic' settings for content authors largely because in the early days they didn't realise that this wasn't a global concept.  It would be good to make it clear that something like italics are not a traditional feature of Japanese – but also explain what the alternatives are, and explain what evolution has (or has not) taken place for users given that italics settings have become ubiquitous in many applications or translated content.

> 6.2 Grapheme/word segmentation and selection
> Links can be made to the chapter describing requirements for dynamic behavior, as well as topics like line wrapping.

This is also relevant for things like double-click word selection, etc., though i don't know whether that kind of thing is in scope for JLReq-D.

> 7.1 Phrase and section boundaries
> This is a topic related to language writing conventions, not layout, so it is outside the scope of jlreq-d.

Note that it's treated fairly extensively in clreq.  Things such as double-full-width dashes could be mentioned here.

> 7.6 Text decoration & other inline features
> jlreq-d does not differentiate between text decoration and emphasis. It is believed that text decoration not intended for emphasis does not exist in Japanese layout. Therefore, link to the emphasis section.

In most of my orthography descriptions this is treated as a kind of catch-all section, for unusual things that didn't really fit in the previous sections.  In fact, i'm progressively changing the heading these days to "Other text decoration & inline features" to make that clearer.  (Of course, if there aren't any, there is no such heading.)

> When scripts are mixed, not only the baseline position but also relative point size adjustments are necessary. Does the Script Resource page address this?

Yes (when there is something to point to).

> 8.6 Styling initials
> Not relevant to Japanese language.

Fwiw, actually i think may be relevant to the Japanese language. See https://r12a.github.io/scripts/jpan/ja.html#initials for one of many examples i have seen.

> 9.3 Footnotes, endnotes, etc
> If the Script Resource page covers non-inline, but also not page-dependent notes, such as side notes, where is it discussed?

Here (under 'etc'). The main distinction is between inline annotations (such as ruby and warichu) and others.

> 9.5 Forms & user interaction
> Since this does not depend on paged media, why is it in Chapter 9?

Form design and positioning are features of layout (such as vertical text orientation of form fields), in a similar way to table and image layout.  In the script resource categorisation the input and other aspects of form design are kept together with the forms themselves, since the scope of this section is wider than just the form layout.  For example, this section can also include any relevant information about input methods or other aspects of user interaction.

hth

**Reply by kidayasuo** [reply]
> a few observations...
> 
> > 5.4 Letterform slopes, weights, & italics
> 
> I think there are some topics where it is useful to say that a typographic feature is not used – and italicisation is one such. Most western developed applications now provide 'italic' settings for content authors largely because in the early days they didn't realise that this wasn't a global concept. It would be good to make it clear that something like italics are not a traditional feature of Japanese – but also explain what the alternatives are, and explain what evolution has (or has not) taken place for users given that italics settings have become ubiquitous in many applications or translated content.

Yes, we are planning to mention that italics or slanting is not a part of Japanese typography. It will also mention what to do if one must achieve slanting (because slanting has become a ubiquitous way of emphasizing text, though ideally avoided).

It could also mention the use of slanting in phototypesetting, mainly for display type, which is now obsolete. Most likely, we will not, because JLReq/JLReq-d are focused on practicality and are not intended to be history books.

> > 6.2 Grapheme/word segmentation and selection
> > Links can be made to the chapter describing requirements for dynamic behavior, as well as topics like line wrapping.
> 
> This is also relevant for things like double-click word selection, etc., though i don't know whether that kind of thing is in scope for JLReq-D.

Yes, we’ll include a section to cover text behavior that didn’t exist in static print, as there are aspects unique to Japanese. These are also important for accessibility, which is one of the focuses of JLReq-D.


> > 7.1 Phrase and section boundaries
> > This is a topic related to language writing conventions, not layout, so it is outside the scope of jlreq-d.
> 
> Note that it's treated fairly extensively in clreq. Things such as double-full-width dashes could be mentioned here.

Language writing conventions are covered by the ‘Council for Japanese Language Policy.’ This is a separate and broad topic, and it has nothing to do with Japanese text layout or CSS.

We will mention double-full-width dashes because you need to use two unicode characters to make it right, and because you can't fold lines between them. (it really should be single unicode character……)


> > 7.6 Text decoration & other inline features
> > jlreq-d does not differentiate between text decoration and emphasis. It is believed that text decoration not intended for emphasis does not exist in Japanese layout. Therefore, link to the emphasis section.
> 
> In most of my orthography descriptions this is treated as a kind of catch-all section, for unusual things that didn't really fit in the previous sections. In fact, i'm progressively changing the heading these days to "Other text decoration & inline features" to make that clearer. (Of course, if there aren't any, there is no such heading.)

Can you think of text decorations or inline features that are not emphasis or inline annotations, and have some Japanese language-specific (or non-Latin) nature or behavior?


> > When scripts are mixed, not only the baseline position but also relative point size adjustments are necessary. Does the Script Resource page address this?
> 
> Yes (when there is something to point to).

👍 It is necessary when you are mixing Japanese and Latin characters from different fonts.

 
> > 8.6 Styling initials
> > Not relevant to Japanese language.
> 
> Fwiw, actually i think may be relevant to the Japanese language. See https://r12a.github.io/scripts/jpan/ja.html#initials for one of many examples i have seen.

The occurrence of examples does not mean such features are common for a particular language.

> > 9.3 Footnotes, endnotes, etc
> > If the Script Resource page covers non-inline, but also not page-dependent notes, such as side notes, where is it discussed?
> 
> Here (under 'etc'). The main distinction is between inline annotations (such as ruby and warichu) and others.

Thank you for the clarification. Non-inline annotations such as side notes (in horizontal text) can be linked from here.

> > 9.5 Forms & user interaction
> > Since this does not depend on paged media, why is it in Chapter 9?
> 
> Form design and positioning are features of layout (such as vertical text orientation of form fields), in a similar way to table and image layout. In the script resource categorisation the input and other aspects of form design are kept together with the forms themselves, since the scope of this section is wider than just the form layout. For example, this section can also include any relevant information about input methods or other aspects of user interaction.

OK, now I understand the intention. Thank you. Could the title of chapter 9 be misleading then?

**Reply by xfq** [reply]
> We will mention double-full-width dashes because you need to use two unicode characters to make it right, and because you can't fold lines between them. (it really should be single unicode character……)

Note that there's a `U+2E3A TWO-EM DASH [⸺]` character. I don't know if this code point was designed for CJK, but unfortunately most Chinese input methods still input two two adjacent `U+2014 EM DASH [—]` characters (with `shift + hyphen`).

> Can you think of text decorations or inline features that are not emphasis or inline annotations, and have some Japanese language-specific (or non-Latin) nature or behavior?

I don't know about Japanese, but for Chinese, maybe the Chinese proper noun marks (underline) and book title marks (only for the type A wavy low line): https://www.w3.org/TR/clreq/#handling_interlinear_punctuation

**Reply by kidayasuo** [reply]
> I don't know about Japanese, but for Chinese, maybe the Chinese proper noun marks (underline) and book title marks (only for the type A wavy low line):

We do not have them or similar marks in Japanese writing conventions. Proper nouns in Kanji are naturally highlighted by Kana characters that surround them. Complex proper nouns or book titles are typically placed within parentheses or quotation marks (like they are in English).
